src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php

```diff
@@ -24,19 +26,42 @@ class WebProfilerExtension extends \Twig_Extension
      * @var ValueExporter
      */
     private $valueExporter;
+    /**
+     * @var HtmlDumper
+     */
+    private $dumper;
+
+    public function __construct(HtmlDumper $dumper = null)
+    {
+        $this->dumper = $dumper ?: new HtmlDumper();
+    }
 
     /**
      * {@inheritdoc}
      */
     public function getFunctions()
     {
         return array(
-            new \Twig_SimpleFunction('profiler_dump', array($this, 'dumpValue')),
+            new \Twig_SimpleFunction('profiler_dump', array($this, 'dumpValue'), array('is_safe' => array('html'))),
         );
     }
 
     public function dumpValue($value)
     {
+        if ($value instanceof Data) {
+            $dump = fopen('php://memory', 'r+b');
+            $prevOutput = $this->dumper->setOutput($dump);
+
+            $this->dumper->dump($value);
+
+            $this->dumper->setOutput($prevOutput);
+            rewind($dump);
+
+            return stream_get_contents($dump);
```

Comment:
This snippet can be reduced to three lines by using the second arg of the dump method and the third of steam_get_contents